server/core/basic_cluster.go

```diff
@@ -25,21 +27,48 @@ import (
 	"go.uber.org/zap"
 )
 
+// SelectConfig provides parameters for selecting regions.
+type SelectConfig struct {
+	NewRegionFirst bool
+	NewProbability float64
+	MaxRegionCount int
+	TimeThreshold  uint64
+}
+
 // BasicCluster provides basic data member and interface for a tikv cluster.
 type BasicCluster struct {
 	sync.RWMutex
-	Stores  *StoresInfo
-	Regions *RegionsInfo
+	Stores     *StoresInfo
+	Regions    *RegionsInfo
+	NewRegions *regionCache
+	SelectConf *SelectConfig
 }
 
 // NewBasicCluster creates a BasicCluster.
 func NewBasicCluster() *BasicCluster {
 	return &BasicCluster{
-		Stores:  NewStoresInfo(),
-		Regions: NewRegionsInfo(),
+		Stores:     NewStoresInfo(),
+		Regions:    NewRegionsInfo(),
+		NewRegions: newRegionCache(),
+		SelectConf: NewSelectConfig(),
+	}
+}
+
+// NewSelectConfig creates a SelectConfig.
+func NewSelectConfig() *SelectConfig {
+	return &SelectConfig{
+		NewRegionFirst: true,
+		NewProbability: 0.5,
+		MaxRegionCount: 1000,
+		TimeThreshold:  60 * 10,
 	}
 }
 
+// SetSelectConfig sets the SelectConfig for selecting regions.
+func (bc *BasicCluster) SetSelectConfig(selectConf *SelectConfig) {
```

Comment:
Is it only used in tests?

Reply:
Yes. This is also due to the potential need for dynamic modification.

Reply:
Yes. This is also due to the potential need for dynamic modification.

Reply:
I'm sorry. Because of a lot of "benchmark" information recently, I've been ignoring some emails until now.